src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/MemoryScope.java

```diff
@@ -171,18 +246,19 @@ MemoryScope acquire() {
         }
 
         @Override
-        MemoryScope dup() { // always called in owner thread
-            return closeOrDup(false);
+        MemoryScope dup(Thread owner) {
+            Objects.requireNonNull(owner, "owner");
```

Comment:
Uhm - not sure about this. Let's say that you have an unchecked segment - you start off unconfined - but then, after some processing, you want to kill the segment and obtained a confined view. Seems a legitimate use case? Although I agree that, in that case, we'd need at least some sort of atomic handshake on the closed bit, to ensure the update either fails or succeeds across multiple racy threads.